bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -563,6 +583,79 @@ func (r *KubeadmConfigReconciler) joinControlplane(ctx context.Context, scope *S
 	return ctrl.Result{}, nil
 }
 
+// resolveProxyUrl adds the user information to the proxy url
+func resolveProxyUrl(proxyUrl string, secret *corev1.Secret) (string, error) {
+	if secret == nil {
+		return proxyUrl, nil
+	}
+	username := secret.StringData["username"]
+	password := secret.StringData["password"]
```

Comment:
What if the secret is malformed (without such fields). Should we have some validation and eventually return some meaningful error messages?